cosmwasm/contract.ts

```diff
@@ -0,0 +1,433 @@
+            );
+        }
+
+        gatewayConfig.codeId = gatewayCodeId || gatewayConfig.codeId;
+        votingVerifierConfig.codeId = verifierCodeId || votingVerifierConfig.codeId;
+        multisigProverConfig.codeId = proverCodeId || multisigProverConfig.codeId;
+    }
+
+    const coordinator = new CoordinatorManager(config);
+    const message = coordinator.constructExecuteMessage(chainName, salt, admin);
+    const msg = [JSON.stringify(message)];
+
+    if (!options.title || !options.description) {
+        options.title = options.title || `Instantiate chain contracts for ${chainName}`;
+        options.description =
+            options.description || `Instantiate Gateway, VotingVerifier and MultisigProver contracts for ${chainName} via Coordinator`;
+    }
+
+    // Need to save deployment info to config, so we can't use executeContractMessage
+    // Handle direct execution and proposal submission separately
+    if (usesGovernanceBypass(config, 'Coordinator')) {
+        if (!confirmDirectExecution(options, msg, coordinatorAddress)) {
+            return;
+        }
+        await executeDirectly(client, coordinatorAddress, msg, fee);
+    } else {
+        await execute(
+            client,
+            config,
+            {
+                ...options,
+                contractName: 'Coordinator',
+                msg,
+            },
+            undefined,
+            fee,
+        );
+    }
+
+    if (!config.axelar.contracts.Coordinator.deployments) {
+        config.axelar.contracts.Coordinator.deployments = {};
+    }
+    config.axelar.contracts.Coordinator.deployments[chainName] = {
+        deploymentName: message.instantiate_chain_contracts.deployment_name,
+        salt: salt,
+    };
+};
+
+const programHandler = () => {
+    const program = new Command();
+
+    program.name('contract').description('Execute contract operations');
+
+    const registerItsChainCmd = program
+        .command('its-hub-register-chains')
+        .description('Register or update an InterchainTokenService chain')
+        .argument('<chains...>', 'list of chains to register or update on InterchainTokenService hub')
+        .addOption(
+            new Option(
+                '--its-msg-translator <itsMsgTranslator>',
+                'address for the message translation contract associated with the chain being registered or updated on ITS Hub',
+            ),
+        )
+        .addOption(
+            new Option(
+                '--its-edge-contract <itsEdgeContract>',
+                'address for the ITS edge contract associated with the chain being registered or updated on ITS Hub',
+            ),
+        )
+        .addOption(new Option('--update', 'update existing chain registration instead of registering new chain'))
+        .action((chains, options) => {
+            options.chains = chains;
```

Comment:
should avoid adding back to `options`. they should be passed in as args separately
https://github.com/axelarnetwork/axelar-contract-deployments/blob/main/evm/its.js#L937